tests/test_pay.py

```diff
@@ -2512,3 +2512,27 @@ def serialize_payload(n):
                      payment_hash=inv['payment_hash'])
 
     l1.rpc.waitsendpay(payment_hash=inv['payment_hash'])
+    invs = l4.rpc.listinvoices(label="lbl")['invoices']
+    assert(len(invs) == 1 and invs[0]['status'] == 'paid')
+
+    pays = l1.rpc.listsendpays()['payments']
+    assert(len(pays) == 1 and pays[0]['status'] == 'complete' and
+           pays[0]['payment_hash'] == inv['payment_hash'])
+
+    # And now for a failing payment, using a payment_hash that doesn't match an
+    # invoice
+    payment_hash = "00"*32
+    onion = l1.rpc.createonion(hops=hops, assocdata=payment_hash)
+    l1.rpc.sendonion(onion=onion['onion'], first_hop=first_hop,
+                     payment_hash=payment_hash)
+
+    try:
+        l1.rpc.waitsendpay(payment_hash=payment_hash)
+        raise ValueError()
```

Comment:
The `try`-block catches `RpcError`s, so we need to use something else to report an error we didn't expect. This is a bit weird here, since we expect this to fail, and not failing is a test-failure. Really anything other than an `RpcError` (or a subclass) would be good here, and `ValueError` is just my goto "don't know what I should use"-error :-)

Reply:
Or the `else:` block can be used but it's strictly the same behaviour.